libparsec/crates/types/src/local_manifest/folder.rs

```diff
@@ -203,7 +203,10 @@ impl LocalFolderManifest {
                     actually_updated = true;
                 }
                 // Add new entry
-                self.children.insert(name, entry_id);
+                if let Some(previous_id) = self.children.insert(name, entry_id) {
+                    // Remove previous id from local confinement points if it was there
+                    self.local_confinement_points.remove(&previous_id);
+                };
```

Comment:
That step is already done at the lines 183-187 (removing children from the map & local confinement points), why it's needed again ?

Reply:
Very good catch, I read too fast and thought the the steps 183-187 were only done for explicitly removed entries. I'll just remove this bit.